src/db/migrations/21-add-heds-collab.ts

```diff
@@ -0,0 +1,31 @@
+import { Knex } from 'knex';
+
+import { NftFactory, NFTContractTypeName, NFTStandard } from '../../types/nft';
+import { MusicPlatform, MusicPlatformType } from '../../types/platform';
+import { addNftFactory, addPlatform, removeNftFactory, removePlatform } from '../migration-helpers';
+
+
+const HEDS_COLLAB_PLATFORM: MusicPlatform =
+  { 
+    id: 'heds-collab',
```

Comment:
hmm is there any easy way we can have these nfts still use heds as the platform instead of introducing a new platform? i guess we'd need to put the custom type onto heds too then, applying only on a specific contract address, or something like that?

Reply:
or maybe cleaner - we just remove platform from the base NftFactory entirely and insert it later on? may not be super easy tho?

Reply:
Not too sure about that because the existing heds platform has a type of `single-track-multiprint-contract` which is used by a few different platforms so having stuff specific to one heds contract in there seems like a bad idea. 

Maybe we can add an optional platform-type field to nftFactory where we can override the platform type from the platform. That way this can still belong to the heds platform but we can process the tracks using the heds-collab platform type?

Reply:
yeh this sounds good!

Reply:
so to use a platform type override you need to specify the id of the platform of the type you want to use. I needed a reference to the platform and not just the platform type, but I think it's safer this way anyway because it's a fk reference and not just a random string. Things might be a bit confusing around platforms / platform types and the naming so let me know if you have any suggestions for improvement @musnit

Reply:
hm ye will look through/think about tmrw

Reply:
hmm it seems kinda weird to add an extra platform that's just used to hack into the type detection system. maybe better to modify the type mapper system - i'll take a look into it now

Reply:
yeah so i think the right way for us to do this would be to modify the processPlatformTracks processor so that it natively supports overriding the type at the nftFactory level via a typeOverride, rather than adding this extra duplicate platform thing

there's some complexities/refactor that will be needed for it, so maybe we should just release this PR under a different platform as you did originally and do that refactor in a later PR?

Reply:
@Geimaj i just took a quick stab at it, incomplete tho: https://github.com/spinamp/spindexer/pull/63